cli/napi/js_native_api.rs

```diff
@@ -781,17 +783,19 @@ fn napi_define_properties(
     let configurable = property.attributes & napi_configurable != 0;
 
     if property.getter.is_some() || property.setter.is_some() {
-      let local_getter: v8::Local<v8::Value> = if property.getter.is_some() {
-        unsafe {
-          create_function(env_ptr, None, property.getter, property.data).into()
-        }
+      let local_getter: v8::Local<v8::Value> = if let Some(getter) =
+        property.getter
+      {
+        create_function(&mut env.scope(), env_ptr, None, getter, property.data)
```

Comment:
If `create_function` accepted `&mut Env` instead of `env_ptr` I think Rust would complain here about double mutable borrow. Should we change `create_function` signature?

Reply:
i would enjoy to be wrong about this but the reason i did this is because i'm not sure its possible to encode the correct borrow in rust's type system for a function which accepts `&mut Env` and uses `env.scope()` to create and return a local handle.

Reply:
Yeah, that might get tricky. That's fine then